Test/domains/urcu-bp.hpp

```diff
@@ -3,22 +3,23 @@
 #include "rcu_domain.hpp"
 
 #include <urcu-bp.h>
-#define _LGPL_SOURCE
```

Comment:
In this branch I attempted to "fix" the preprocessor stuff here by moving `#define _LGPL_SOURCE` up above the `#include <urcu-bp.h>` that it's clearly supposed to affect; but when I did that, I got compiler errors like

```
/usr/local/Cellar/userspace-rcu/0.9.1/include/urcu/static/wfcqueue.h:384:10: error: cannot initialize return
      object of type 'struct cds_wfcq_node *' with an rvalue of type 'void *'
                return CDS_WFCQ_WOULDBLOCK;
                       ^~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/userspace-rcu/0.9.1/include/urcu/wfcqueue.h:46:29: note: expanded from macro
      'CDS_WFCQ_WOULDBLOCK'
#define CDS_WFCQ_WOULDBLOCK     ((void *) -1UL)
                                ^~~~~~~~~~~~~~~
```

so I punted on that issue by just removing the useless `#define` lines.